src/qibo/backends/numpy.py

```diff
@@ -524,7 +524,7 @@ def execute_circuit_repeated(self, circuit, nshots, initial_state=None):
 
         if circuit.density_matrix:  # this implies also it has_collapse
             assert circuit.has_collapse
-            final_state = np.asarray(final_states).mean(0)
+            final_state = self.np.mean(self.to_numpy(final_states), 0)
```

Comment:
Since you are already casting to numpy, I guess you could do the `mean` in numpy
```suggestion
            final_state = np.mean(self.to_numpy(final_states), 0)
```